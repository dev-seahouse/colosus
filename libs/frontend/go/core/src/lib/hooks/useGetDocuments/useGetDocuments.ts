import { QueryClient, useQuery } from '@tanstack/react-query';
import { ConnectInvestorLegalDocumentApi } from '@bambu/api-client';
import type { ConnectInvestorGetDocumentsResponseDto } from '@bambu/api-client';

export type GetDocumentsData = ConnectInvestorGetDocumentsResponseDto;

export const getDocumentsQuery = () => ({
  queryKey: ['getDocuments'],
  queryFn: async () => {
    const connectInvestorLegalDocumentApi =
      new ConnectInvestorLegalDocumentApi();
    const res = await connectInvestorLegalDocumentApi.getLegalDocuments();

    return res.data;
  },
  staleTime: Infinity,
});

export const getDocumentsLoader =
  (queryClient: QueryClient) => async (): Promise<GetDocumentsData> => {
    const query = getDocumentsQuery();

    // ⬇️ return data or fetch it
    return (
      queryClient.getQueryData(query.queryKey) ??
      (await queryClient.fetchQuery(query))
    );
  };

export interface UseGetDocumentsOptions {
  initialData?: GetDocumentsData;
}

/**
 * hook to get tenant documents data
 */
export const useGetDocuments = ({
  initialData,
}: UseGetDocumentsOptions = {}) => {
  return useQuery({ ...getDocumentsQuery(), initialData });
};

export default useGetDocuments;
