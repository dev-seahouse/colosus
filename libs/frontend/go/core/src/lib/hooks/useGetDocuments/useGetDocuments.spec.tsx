import {
  renderHook,
  waitFor,
  queryClientWrapper,
} from '@bambu/react-test-utils';

import useGetDocuments from './useGetDocuments';

describe('useGetDocuments', () => {
  it('should render successfully', async () => {
    const { result } = renderHook(() => useGetDocuments(), {
      wrapper: queryClientWrapper,
    });

    await waitFor(() => expect(result.current.isSuccess).toBe(true));

    expect(result.current.data).toMatchSnapshot();
  });
});
