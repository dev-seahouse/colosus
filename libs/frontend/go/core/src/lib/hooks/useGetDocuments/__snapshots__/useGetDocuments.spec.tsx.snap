// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`useGetDocuments > should render successfully 1`] = `
{
  "privacyPolicyUrl": "some-pdf.pdf",
  "termsAndConditionsUrl": "some-pdf.pdf",
}
`;
